verity/src/helm/templates/_helpers.tpl

```diff
@@ -61,3 +91,27 @@ Vault injection template
 {{ end }}
 {{- end }}`}}
 {{- end }}
+
+{{/*
+Vault annotations template
+*/}}
+{{- define "verity.vaultAnnotations" -}}
+{{- $credentials := printf "kubernetes_secrets/eks-%s/%s-%s" .Values.gitlab_env .Values.name (include "verity.service" .) }}
+{{- $credentials_tf := printf "kubernetes_secrets/eks-%s/%s-%s-tf" .Values.gitlab_env .Values.name (include "verity.service" .) }}
+{{- $app_confluent := printf "kubernetes_secrets/eks-%s/%s/app-confluent" .Values.gitlab_env .Values.name }}
+vault.hashicorp.com/agent-inject: "true"
+vault.hashicorp.com/role: "k8s-eks-{{ coalesce .Values.vault_auth_role .Values.env }}"
+vault.hashicorp.com/auth-path: "auth/k8s/eks-{{ coalesce .Values.vault_auth_path .Values.gitlab_env }}"
+vault.hashicorp.com/agent-inject-secret-credentials: "{{ $credentials }}"
+vault.hashicorp.com/agent-inject-secret-credentials-tf: "{{ $credentials_tf }}"
+vault.hashicorp.com/agent-inject-secret-app-confluent: "{{ $app_confluent }}"
+vault.hashicorp.com/tls-secret: vault-tls-secret
+vault.hashicorp.com/ca-cert: /vault/tls/evernym-root-ca.crt
+vault.hashicorp.com/agent-pre-populate-only : "true"
+vault.hashicorp.com/agent-inject-template-credentials: |
+{{ include "verity.vaultInjectTemplate" (dict "Secret" $credentials) | indent 2 }}
+vault.hashicorp.com/agent-inject-template-credentials-tf: |
+{{ include "verity.vaultInjectTemplate" (dict "Secret" $credentials_tf) | indent 2 }}
+vault.hashicorp.com/agent-inject-template-app-confluent: |
+{{ include "verity.vaultInjectTemplate" (dict "Secret" $app_confluent) | indent 2 }}
+{{- end }}
```

Comment:
I think this code makes our codebase inconsistent and difficult to understand.
We decided to use AWS Secrets manager instead of Hashicorp vault. To support AWS Secrets we changed the code (entrypoint.sh) in the docker image
old
```
VAULT_SECRETS=("/vault/secrets/credentials-tf" "/vault/secrets/credentials" "/vault/secrets/app-confluent")
 

 # Source secret files injected from Vault (if they exist)
 for f in "${VAULT_SECRETS[@]}" ; do
     if [ -f $f ]; then
         source $f
         # rm $f
     fi
 done
```

new
```
 VAULT_SECRETS=("/vault/secrets/credentials-tf.json" "/vault/secrets/credentials.json" "/vault/secrets/app-confluent.json")

 # Source secret files injected from Vault (if they exist)
 for f in "${VAULT_SECRETS[@]}" ; do
     if [ -f $f ]; then
         tmp_conf_file=$(mktemp /tmp/XXXXXXXX)
         python3 /usr/local/bin/json_to_config.py -j $f -c $tmp_conf_file
         source $tmp_conf_file
         rm $tmp_conf_file
     fi
 done
```

Above code don't support backward compatibility with Hashicorp Vault. I think we have either to remove the Vault hashicorp section from `_helpers.tpl` or to implement backward compatibility in the `entrypoint.sh`

Reply:
addressed